CompositionRoot/B.swift

```diff
@@ -1,17 +1,17 @@
-protocol BPrototol {
+protocol BProtocol {
     func bMethod()
 }
 
-class B: BPrototol {
+class B: BProtocol {
     
-    let applicationContext: ApplicationContext
+    let cFactory: CFactoryProtocol
     
-    init(applicationContext: ApplicationContext) {
-        self.applicationContext = applicationContext
+    init(cFactory: CFactoryProtocol) {
+        self.cFactory = cFactory
     }
     
     func bMethod() {
-        let c = C(applicationContext: applicationContext)
```

Comment:
1. B does not depend on anything in `ApplicationContext`.  But we have to inject it into `B`.  It's confusing. 
2. `A` use `B`,  `B` use `C`,  `C` uses `D`.  `D` uses many properties in `ApplicationContext`.  `B` and `C` does not use any property in `ApplicationContext`, but we have to inject `ApplicationContext` into `B` and `C` just because `D` use it.   
3. This is dependency transitive.  It transit all of the dependencies from low level classes to the high level classes. This make the top levels depends on all of the dependencies of all low level classes. 
We can fix it by using `Composition Root` pattern. See the right side.